src/FormCheck.tsx

```diff
@@ -152,7 +153,9 @@ const FormCheck: BsPrefixRefForwardingComponent<'input', FormCheckProps> =
         [controlId, id],
       );
 
-      const hasLabel = label != null && label !== false && !children;
+      const hasLabel =
+        (!children && label != null && label !== false) ||
+        hasChildOfType(children, FormCheckInput);
```

Comment:
Should this be `FormCheckLabel` instead of `FormCheckInput`?

Reply:
Whoops, good catch!

Reply:
[`f0d8e9f`](https://github.com/react-bootstrap/react-bootstrap/pull/5983/commits/f0d8e9f04b70d248a17c0f76d381ea6bded8c9c5)